core/sdk-core/src/main/java/software/amazon/awssdk/core/retry/RetryPolicy.java

```diff
@@ -54,7 +55,12 @@ private RetryPolicy(BuilderImpl builder) {
         this.throttlingBackoffStrategy = builder.throttlingBackoffStrategy;
         this.numRetries = builder.numRetries;
         this.retryConditionFromBuilder = builder.retryCondition;
-        this.retryCondition = AndRetryCondition.create(MaxNumberOfRetriesCondition.create(numRetries), retryConditionFromBuilder);
+        if (numRetries == 0) {
```

Comment:
Hmm why do we do this?

Reply:
`RetryCondition.none()` would be `MaxNumberOfRetriesCondition(maxNumberOfRetries=0)`
If we don't add the check here, it will add another duplicate condition `MaxNumberOfRetriesCondition(maxNumberOfRetries=0)`

`RetryPolicy.none()` would resolve to 
```
AndRetryCondition(conditions=[MaxNumberOfRetriesCondition(maxNumberOfRetries=0), MaxNumberOfRetriesCondition(maxNumberOfRetries=0)])
```

Alternately, we can update `AndRetryCondition` to a Set instead of a list to fix this.

Reply:
hmm, seems like it will introduce another bug. 
`RetryPolicy.builder().numRetris(0).build()` would still retry in this case.

I'll update the PR and use Set in `AndReryCondition` instead